product_variant_storage_location/product.py

```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+# © 2016 Alex Comba - Agile Business Group
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    loc_rack = fields.Char(string='Rack', size=16)
+    loc_row = fields.Char(string='Row', size=16)
+    loc_case = fields.Char(string='Case', size=16)
+
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    @api.multi
+    def write(self, vals):
+        """Propagate to the variants the template storage location data
+        (if modified)."""
+        res = super(ProductTemplate, self).write(vals)
+        if ('loc_rack' in vals):
+            self.mapped('product_variant_ids').write(
```

Comment:
Don't write three times. Add the values to a dictionary and if the dictionary contains any value, then write.